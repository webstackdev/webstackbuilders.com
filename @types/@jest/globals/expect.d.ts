import type { Constructor } from '../../../test/jest/matchers/assertions'

declare module 'expect' {
  export interface Matchers<R> {
    /**
     * Custom Jest matchers
     */
    toHaveInProtoChain(...chain: Constructor[]): R
  }
}
