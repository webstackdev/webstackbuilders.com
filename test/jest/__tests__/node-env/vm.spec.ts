import { describe, expect, test } from '@jest/globals'
import { loadHtmlTemplate, tsCompile } from '../../helpers/vm'
import { resolve } from 'path'

const getFixturePath = (fileName: string) => {
  return resolve(__dirname, '../../__fixtures__/compileTs', fileName)
}

describe(`tsCompile compiles inline Typescript script`, () => {
  test(`compiles valid Typescript`, async () => {
    const fixturePath = getFixturePath(`compileTs_1.ts`)
    //expect(sut).toEqual(expect.stringContaining(`const foo = (input) => { return input; };`))
    expect(await tsCompile(fixturePath)).toMatchInlineSnapshot(`
      "/******/ (function() { // webpackBootstrap
      /******/ 	"use strict";
      var __webpack_exports__ = {};
      /*!*********************************************************!*\\
        !*** ./test/jest/__fixtures__/compileTs/compileTs_1.ts ***!
        \\*********************************************************/

      const foo = (input) => { return input; };
      foo(\`test\`);

      /******/ })()
      ;"
    `)
  })
})

describe('loadHtmlTemplate', () => {
  const templatePath = resolve(`test/jest/__fixtures__/loadHtmlTemplate.njk`)
  test('test', async () => {
    expect(await loadHtmlTemplate(templatePath)).toMatchSnapshot()
  })
})
