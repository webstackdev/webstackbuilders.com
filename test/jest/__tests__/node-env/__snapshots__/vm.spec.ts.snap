// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadHtmlTemplate test 1`] = `"<p id="test__paragraph">This is a <a id="test__anchor" href="/test">test template</a>.</p>"`;
